akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphTransposeSpec.scala

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (C) 2014-2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.testkit._
+import akka.stream.testkit.Utils._
+import akka.stream._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class GraphTransposeSpec extends TwoStreamsSetup {
+  import GraphDSL.Implicits._
+
+  override type Outputs = Seq[Int]
+
+  override def fixture(b: GraphDSL.Builder[_]): Fixture = new Fixture(b) {
+    val transpose = b.add(Transpose[Int](2))
+
+    override def left: Inlet[Int] = transpose.in(0)
+    override def right: Inlet[Int] = transpose.in(1)
+    override def out: Outlet[Seq[Int]] = transpose.out
+  }
+
+  "Transpose" must {
+
+    "work in the happy case" in assertAllStagesStopped {
+      val probe = TestSubscriber.manualProbe[Seq[Int]]()
+
+      RunnableGraph.fromGraph(GraphDSL.create() { implicit b ⇒
+        val transpose = b.add(Transpose[Int](2))
+
+        Source(1 to 4) ~> transpose.in(0)
+        Source(2 to 5) ~> transpose.in(1)
+
+        transpose.out ~> Sink.fromSubscriber(probe)
+
+        ClosedShape
+      }).run()
+
+      val subscription = probe.expectSubscription()
+
+      subscription.request(2)
+      probe.expectNext(Seq(1, 2))
+      probe.expectNext(Seq(2, 3))
+
+      subscription.request(1)
+      probe.expectNext(Seq(3, 4))
+      subscription.request(1)
+      probe.expectNext(Seq(4, 5))
+
+      probe.expectComplete()
+    }
+
+    "complete if one side is available but other already completed" in {
+      val upstream1 = TestPublisher.probe[Int]()
+      val upstream2 = TestPublisher.probe[Int]()
+
+      val completed = RunnableGraph.fromGraph(GraphDSL.create(Sink.ignore) { implicit b ⇒
```

Comment:
Use a probe as Sink and assert that the `Seq(1,2)` is properly signaled before completion triggers?